### DISCUSSION TITLE
How to merge code coverage report between vitest and Cypress?

### DISCUSSION THREAD

**Discussion by gweesin** [vitest-dev/vitest#3975] [Q&A] [answered]
I found our website said that:
> If you'd like to use Cypress, we suggest using Vitest for all headless logic in your application and Cypress for all browser-based logic.

I think it's a perfect idea, and I really do like this in my UI library! But I want to generate code coverage report with CI, and judge errors if the percent less than 80%. so how can I merge code coverage report between vitest and Cypress?

**Comment by AriPerkkio** [comment]
Which CI are you using?

I've used this same exact setup for a component library - Cypress Component testing for browser stuff and Vitest for headless logic. It worked really well and code coverage was merged by the CI. I used istanbul instrumentation on Cypress using `vite-plugin-istanbul` + `@cypress/code-coverage` and on Vitest the `@vitest/coverage-istanbul`. 

The CI service I used was able to merge multiple `lcov` reports by itself.

**Reply by gweesin** [reply]
Thanks for your answer. I'm using the Jenkins tool to build my CI pipeline, but I haven't found how to merge test reports using Jenkins. Which CI tools are you using?

**Reply by AriPerkkio** [reply, marked as answer]
I don't think I was using Jenkins back then. But if your CI does not support reading multiple coverage reports, you can merge the reports with https://github.com/istanbuljs/nyc#combining-reports-from-multiple-runs

**Reply by hammzj** [reply]
2 years late to this comment, but I want to report that this does not always work correctly and can produce inaccurate results at times, if there are Cypress tests that check `.ts/.js` as well, in the case of things like Vue composables that need a wrapped component.

In this case where Cypress instruments against both `.vue` component and `.ts` files, it won't detect that Vitest is covering other `.ts` files and report a lot of missed coverage on those files. When combined with an accurate Vitest output, the finalized report will be completely inaccurate. Component tests against `.vue` files are fine, but the `.ts` files show incorrect results; it even checks against JSDoc comments after merging, which makes no sense either.
